WordPress/gutenberg
# Post Editor: Always force iframe in pattern editor

Noticed this while reviewing #65885

## What?

In the post editor, force the pattern editor to always be an iframe, regardless of whether the main editor is an ifarme or not. This will make the pattern editor's height fit its content.

| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/c0de7ba8-23c9-4734-9abb-348f7bc1bd9b)| ![image](https://github.com/user-attachments/assets/14980518-3bed-46a4-af43-fb374e73e15d) | 

## Why?

Whether the pattern editor in the post editor is an iframe or not is determined by whether the main editor is an iframe or not.

However, the pattern editor is a resizable editor and the editor height is expected to fit the content. Additionally, the padding appender (the functionality that adds a new block when you click below the last block) is not expected to be used with the pattern editor.

Therefore, if the main editor is not an iframe, clicking below the content in the pattern editor does nothing.

## How?

Add a post type check.

However, changing the conditions that determine if it should be an iframe in a lower component (`VisualEditor`) may not be ideal - whether it should be an iframe or not is something the post editor should only care about. In the future, it may be possible to aggregate these conditions at a higher layer ([`useShouldIframe()` hook](https://github.com/WordPress/gutenberg/blob/9d67345a00a0ce3f1ca40b5894e45fc3b9088857/packages/edit-post/src/components/layout/use-should-iframe.js#L32-L37)).

**Update:** I consolidated all the conditions that determine whether to run an iframe or not into the useShouldIframe hook.

## Testing Instructions

- To run the editor as a non-iframe, you need to set it up as follows:
  - Enable a classic theme.
  - Register a block whose API version is less than or equal to 2. For example, enable [the CoBlocks plugin](https://wordpress.org/plugins/coblocks/).
- Open the post editor.
- Make sure the editor is NOT an iframe.
- Create a synced pattern and click the "Edit original" button.
- Make sure the pattern editor is an iframe.
